Updated README multiple servers examples and added some cosmetic touch-ups

viorelsfetea opened the pull request:
Hi,

I had a problem with the deploy to multiple servers example, because the properties, such as containerId were not added to a `cargo {}` block when the tasks were automatically created. This resulted in an error saying "Container ID not set."

Since I was there, I also did some cosmetic touch-ups, such as:
1. Added a table of contents
2. Updated the code blocks to highlight groovy code

bmuschko commented on README.md:
```diff
@@ -439,12 +488,14 @@ also add another task that triggers the deployment to all remote containers.
-            port = config.port
-            username = config.username
-            password = config.password
+            cargo {
```
Here you are setting the property values for the extension and not the task type. Setting extension property values vs. task type values are two different concept. Mixing them up in this sample only works accidentally and will be confusing to users.

My guess is that you are applying the `cargo` plugin? For this specific use case you should be applying the `cargo-base` plugin. Maybe that's not fully clear from the readme file. Can you provide me with the full build script?

viorelsfetea commented:
Yes, you are absolutely right. I must admit, I was totally confused here as to what the difference was since the block was just creating gradle tasks and the type of each task was referring to `com.bmuschko.gradle.cargo`.

Thank you for clarifying. I've removed the cargo{} block and I've added a note. Please merge only if you consider the other changes useful. Otherwise, please let me know and I'll close the PR.

viorelsfetea commented:
I'll just squash now the commits so it's cleaner

bmuschko commented:
@viorelsfetea Just had another look. The indentation of the code examples is off in the rendered output. They now have an additional tab/4 spaces. We'll want to have them start at the beginning of the grey box. Would you mind fixing that so I can merge?

bmuschko commented:
Looking at this again. I don't think there's much benefit to adding a TOC manually. The TOC should be automatically generated as part of a user guide similar to what we do [here](https://github.com/bmuschko/gradle-docker-plugin/blob/master/gradle/asciidoc.gradle). It'll be much easier to maintain. Please open a new pull request if you are interested in adding a similar user guide to  the one available for the Gradle Docker plugin. You can find the relevant PR [here](https://github.com/bmuschko/gradle-docker-plugin/pull/630).
